src/utils/eth-utils.ts

```diff
@@ -22,3 +24,71 @@ export const decodeAccountState = (state: Buffer): any => {
     codeHash: toHexString(account.codeHash),
   }
 }
+
+export const waitForNetwork = async (
+  provider: JsonRpcProvider,
+  name: string,
+  logger: any
+): Promise<void> => {
+  logger.info(`Trying to connect to network: ${name}...`)
+
+  for (let i = 0; i < 10; i++) {
+    try {
+      await provider.detectNetwork()
+      logger.info(`Successfully connected to network: ${name}.`)
+      break
+    } catch (err) {
+      if (i < 9) {
+        logger.info(
+          `Unable to connect network: ${name}, will try ${
+            10 - i
+          } more time(s)...`
+        )
+        await sleep(1000)
+      } else {
+        throw new Error(
+          `Unable to connect to network: ${name}, check that your endpoint is correct.`
+        )
+      }
+    }
+  }
+}
+
+export const smartSendTransaction = async ({
+  contract,
+  functionName,
+  functionParams = [],
+  acceptableErrors = [],
+  logger,
+}: {
+  contract: Contract
+  functionName: string
+  functionParams?: any[]
+  acceptableErrors?: Array<{
+    errors: string[]
+    reason: string
+  }>
+  logger?: any
+}): Promise<void> => {
+  try {
+    await await contract[functionName](...functionParams).wait()
+  } catch (err) {
+    try {
+      await contract.callStatic[functionName](...functionParams)
+    } catch (err) {
+      for (const acceptableError of acceptableErrors) {
+        for (const error of acceptableError.errors) {
+          if (err.toString().includes(error)) {
```

Comment:
Should this have another check for the hex string that is sometimes returned? The hex string could be created with `b = Buffer.from(error, 'utf8').toString('hex')`